tcmur_cmd_handler.c

```diff
@@ -144,6 +144,124 @@ static int write_work_fn(struct tcmu_device *dev,
 				block_size * tcmu_get_lba(cmd->cdb));
 }
 
+static void handle_write_same_16_null_cbk(struct tcmu_device *dev,
+					  struct tcmulib_cmd *cmd,
+					  int ret)
+{
+}
+
+static int write_same_16_work_fn(struct tcmu_device *dev,
+				 struct tcmulib_cmd *cmd)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint32_t lba_cnt = tcmu_get_xfer_length(cmd->cdb);
+	uint64_t start_lba = tcmu_get_lba(cmd->cdb);
+	uint64_t end_lba;
+	uint32_t max_lbas, copy_lbas;
+	uint32_t max_sectors;
+	int ret;
+
+	end_lba = start_lba + lba_cnt;
+
+	max_sectors = tcmu_get_attribute(dev, "hw_max_sectors");
+	max_lbas = min(max_sectors, ((uint32_t)(~0U)));
+
+	tcmu_dbg("Start lba: %llu Number of lba:: %hu, end lba: %llu\n",
+		 start_lba, lba_cnt, end_lba);
+
+	/*
+	 * Write contents of the logical block data(from the Data-Out Buffer)
+	 * to each LBA in the specified LBA range.
+	 */
+	while(start_lba < end_lba) {
+		copy_lbas = min(max_lbas, lba_cnt);
+
+		if (start_lba + copy_lbas >= end_lba)
+			cmd->done = handle_generic_cbk;
+
+		ret = rhandler->write(dev, cmd, cmd->iovec, cmd->iov_cnt,
+				      tcmu_iovec_length(cmd->iovec, cmd->iov_cnt),
```

Comment:
Yes it is.

I'll update them both later.

Thanks very much for liuy's advice.